engine-sdk/src/lib.rs

```diff
@@ -41,6 +41,21 @@ pub fn log_utf8(bytes: &[u8]) {
         exports::log_utf8(bytes.len() as u64, bytes.as_ptr() as u64);
     }
 }
+/// Calls random seed produced by validators
+#[cfg(feature = "contract")]
+pub fn random_seed() -> H256 {
+    unsafe {
+        exports::random_seed(0);
+        let bytes = H256::zero();
+        exports::read_register(0, bytes.0.as_ptr() as *const u64 as u64);
+        bytes
+    }
+}
+
+#[cfg(not(feature = "contract"))]
+pub fn random_seed() -> H256 {
+    H256::zero()
+}
```

Comment:
Regardless of whether we pass `random_seed` all the from `engine/src/lib.rs` into core logic, or if you implement it as part of the `Env` trait, I think this implementation should be deleted. We must have feature parity between the wasm engine and the standalone engine otherwise the debug tracing will sometimes return wrong results.

Reply:
Done